### DISCUSSION TITLE
Rebalance in strimzi-kafka-operator be fully automated?

### DISCUSSION THREAD

**Discussion by psamagal** [strimzi/strimzi-kafka-operator#4111] [General]
Hi,

Can the rebalance in strimzi-kafka-operator be fully automated?
Meaning every time we scale up or down, the operator itself, trigger the kafka rebalance operation without the need of manual intervation?

I can using the annotations do a manual rebalance of the cluster, as per instructions in here https://strimzi.io/blog/2020/06/15/cruise-control/.

But can this be done in an automated way? and how?

regards

**Comment by tomncooper** [comment]
We do have plans to integrate CC more into the scaling process but it will be a while before we get around to it. Contributions are welcome of course. You could also create a proposal in the strimzi/proposals repo if you have an idea of how this could be implemented? Obviously we would want this to be optional and how would a user tune the post-scale rebalanced options?

**Reply by scholzj** [reply]
@tomncooper Do you know if we have some enhancement issue for it?

**Reply by tomncooper** [reply]
Not specifically, I should create one as well as one to cover the replica scaling integration.

**Reply by scholzj** [reply]
I'm sure we have something for replica changes. Not sure about the scale-down/up integration.

**Reply by tomncooper** [reply]
Yes we do for the replicas: https://github.com/strimzi/strimzi-kafka-operator/issues/3267

**Comment by tomncooper** [comment]
@psamagal I added an enhancement issue to cover this (https://github.com/strimzi/strimzi-kafka-operator/issues/4112). If you have any additional thoughts or context, you can add them there.

**Reply by psamagal** [reply]
That sounds perfect, will follow it. :D
